### DISCUSSION TITLE
[Idea] Add a hook in webservice to add custom fields

### DISCUSSION THREAD

**Discussion by web-cooking-factory** [PrestaShop/PrestaShop#32039] [Ideas]
### The problem

There's ways to add custom fields to objects. But there's no way to make these fields available through webservice except by creating custom endpoints.

### The solution

Have a hook that allows to add or retrieve custom fields of an object directly.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by HunkBenny** [comment]
I submitted the same idea today; am willing to look into this when I have time. I also believe that this would be a valuable addition
#39380
